src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -604,14 +605,46 @@ public function iShouldNotBeAbleToSelectTheVariant(string $productVariantName):
      * @Then /^I should(?:| also) see the product association "([^"]+)" with (products "[^"]+" and "[^"]+")$/
      */
     public function iShouldSeeTheProductAssociationWithProductsAnd(string $productAssociationName, array $products): void
+    {
+        Assert::true($this->isProductAssociationWithProductsAvailable($productAssociationName, $products));
+    }
+
+    /**
+     * @Then /^I should(?:| also) see the product association "([^"]+)" with (product "[^"]+")$/
+     */
+    public function iShouldSeeTheProductAssociationWithProduct(string $productAssociationName, ProductInterface $product): void
+    {
+        Assert::true($this->isProductAssociationWithProductsAvailable($productAssociationName, [$product]));
+    }
+
+    /**
+     * @Then /^I should(?:| also) not see the product association "([^"]+)" with (product "[^"]+")$/
+     */
+    public function iShouldNotSeeTheProductAssociationWithProduct(string $productAssociationName, ProductInterface $product): void
+    {
+        Assert::false($this->isProductAssociationWithProductsAvailable($productAssociationName, [$product]));
+    }
+
+    /**
+     * @Then /^I should not see the product (association "([^"]+)")$/
+     */
+    public function iShouldNotSeeTheProductAssociation(ProductAssociationTypeInterface $productAssociationType): void
     {
         /** @var ProductInterface $product */
         $product = $this->sharedStorage->get('product');
 
         $response = $this->client->show(Resources::PRODUCTS, $product->getCode());
         $associations = $this->responseChecker->getValue($response, 'associations');
 
-        Assert::true($this->hasAssociationsWithProducts($associations, $productAssociationName, $products));
+        foreach ($associations as $association) {
+            $associationResponse = $this->client->showByIri($association);
+
+            if ($associationResponse->getStatusCode() === 200) {
+                $associationTypeIri = $this->responseChecker->getValue($associationResponse, 'type');
+
+                Assert::notSame($associationTypeIri, $this->iriConverter->getIriFromItem($productAssociationType));
```

Comment:
```suggestion
                Assert::notSame($associationTypeIri, $productAssociationTypeIri);
```
Move the iri conversion before the foreach